### PR TITLE
Fix VLC video playback issues on Raspberry Pi 4. :wrench:

### LINKED ISSUE #1679: The viewer container keeps restarting due to issues related to VLC.
#### Overview of the Issue

The display shows a black screen.

Here's what I got upon checking the viewer container:

```
./bin/start_viewer.sh: line 36: /sys/fs/cgroup/memory/memory.swappiness: No such file or directory
vlc: unknown option or missing mandatory argument `--mmal-display=HDMI-2'
Try `vlc --help' for more information.
Traceback (most recent call last):
  File "viewer.py", line 60, in <module>
    media_player = VLCMediaPlayer() if 'Raspberry Pi 4' in get_raspberry_model() else OMXMediaPlayer()
  File "/usr/src/app/lib/media_player.py", line 35, in __init__
    self.player = self.instance.media_player_new()
AttributeError: 'NoneType' object has no attribute 'media_player_new'
./bin/start_viewer.sh: line 54: kill: (37) - No such process
./bin/start_viewer.sh: line 36: /sys/fs/cgroup/memory/memory.swappiness: No such file or directory
vlc: unknown option or missing mandatory argument `--mmal-display=HDMI-2'
Try `vlc --help' for more information.
Traceback (most recent call last):
  File "viewer.py", line 60, in <module>
    media_player = VLCMediaPlayer() if 'Raspberry Pi 4' in get_raspberry_model() else OMXMediaPlayer()
  File "/usr/src/app/lib/media_player.py", line 35, in __init__
    self.player = self.instance.media_player_new()
AttributeError: 'NoneType' object has no attribute 'media_player_new'
./bin/start_viewer.sh: line 54: kill: (38) - No such process
```

#### Reproduction Steps

```bash
cd ~/screenly
./bin/upgrade_containers.sh # pulls the images and start the containers
docker logs -f screenly-srly-ose-viewer-1
```

#### Environment

* Raspberry Pi Hardware Version: Raspberry Pi 4 Model B
* Raspberry Pi Network Setup: Ethernet
* OS: Raspberry Pi OS Lite (Bullseye), 64-bit

### PR DESCRIPTION
**Pull request by nicomiguelino**
Fixes #1679 (temporarily).

### CODE REVIEW

**Comment by rusko124** [comment]
OMXplayer is deprecated
https://github.com/popcornmix/omxplayer

**Comment by nicomiguelino** [comment]
@rusko124, thanks for pointing that out. Yeah, it doesn't work well with RPi 4. For instance, it can play MP4 videos, but not the MOV ones.

File: ansible/roles/system/tasks/main.yml [hunk 1]
```diff
@@ -73,8 +67,6 @@
   ansible.builtin.debug:
     msg: "Use cmdline.txt.orig for boot parameters (don't remove this file)"
 
-- name: Copy cmdline.txt.orig to cmdline.txt
-  ansible.builtin.copy:
```
Comment (nicomiguelino) [inline ansible/roles/system/tasks/main.yml:77]: I think I've accidentally removed this one.

**Comment by nicomiguelino** [comment]
@vpetersson, I currently do proof-of-concepts by running VLC by hand on `srly-ose-viewer`. I was able to make audio play on my TV screen. There's still no luck in making MP4 playback (at the very least) work inside the container.

**Comment by vpetersson** [comment]
> @vpetersson, I currently do proof-of-concepts by running VLC by hand on `srly-ose-viewer`. I was able to make audio play on my TV screen. There's still no luck in making MP4 playback (at the very least) work inside the container.
> 
> If the fix for VLC will take longer, this PR can be merged to partially and temporarily unblock video playback, although it will still not be able to play `.mov` files.

Well, what is the codecs used in this `.mov` file? You can check this using `ffprobe`. In short, the Raspberry Pi is very picky with the codecs it supports. Anything other than h264 (and h625 IIRC on the Pi 4) is unsupported.

**Comment by nicomiguelino** [comment]
> Well, what is the codecs used in this `.mov` file? You can check this using `ffprobe`. In short, the Raspberry Pi is very picky with the codecs it supports. Anything other than h264 (and h625 IIRC on the Pi 4) is unsupported.

Here's what I got after I ran `ffprobe -v error -select_streams v:0 -show_entries stream=codec_name -of default=noprint_wrappers=1:nokey=1 [path-to-file].mov`: `mpeg4`

On the other hand, running the same command on a `.mp4` file gives: `h264`

**Comment by vpetersson** [comment]
So the file extension doesn't matter. Linux doesn't care about this at all (it's a Windows thing). Based on what you output, this `.mov` file you're using seems broken and should probably be disregarded. I usually use [Big Buck Bunny](https://peach.blender.org/trailer-page/) to test video.

**Comment by vpetersson** [comment]
Oh hold on -- if it worked with `omxplayer`, it should work with `vlc`. My assumption was based that you couldn't get it to work at all.

**Comment by nicomiguelino** [comment]
> Oh hold on -- if it worked with `omxplayer`, it should work with `vlc`. My assumption was based that you couldn't get it to work at all.

Got it. 📝 Since the `--mmal-output` and `--mmal-vout` was removed, I'm still figuring out what flags to use or what other options to take.

**Comment by vpetersson** [comment]
> For reference, I'm also getting the following when running `DISPLAY=:0 sudo -u viewer cvlc tests/assets/asset.mov` inside the viewer container (only the audio's working):
> 
> ```
> error: XDG_RUNTIME_DIR not set in the environment.
> error: XDG_RUNTIME_DIR not set in the environment.
> [f6512708] gl vout display error: parent window not available
> [f6512708] gles2 vout display error: parent window not available
> [f6512708] xcb vout display error: window not available
> [f6512708] xcb vout display error: window not available
> [f6512708] fb vout display error: cannot get terminal mode (Inappropriate ioctl for device)
> [f6512708] caca vout display error: cannot initialize libcaca
> [eee5d6f0] main video output error: video output creation failed
> [f6810e08] main decoder error: failed to create video output
> [mpeg4 @ 0xf680ccc0] get_buffer() failed
> [mpeg4 @ 0xf680ccc0] thread_get_buffer() failed
> [mpeg4 @ 0xf680ccc0] get_buffer() failed (-1 (nil))
> ```
> 
> Here's the output after running `cvlc tests/assets/asset.mov` in host OS:
> 
> Video and audio is working fine.
> 
> ```
> VLC media player 3.0.17.4 Vetinari (revision 3.0.13-8-g41878ff4f2)
> [00000055a8662bd0] vlcpulse audio output error: PulseAudio server connection failure: Connection refused
> [00000055a862d200] main interface error: no suitable interface module
> [00000055a858b5c0] main libvlc error: interface "globalhotkeys,none" initialization failed
> [00000055a862d200] dummy interface: using the dummy interface module...
> [0000007f70001fc0] gles2 generic error: parent window not available
> [0000007f700013e0] mmal_xsplitter vout display error: Failed to open Xsplitter:opengles2 module
> [0000007f70001fc0] xcb generic error: window not available
> [0000007f700013e0] mmal_xsplitter vout display error: Failed to open Xsplitter:xcb_x11 module
> [0000007f700013e0] mmal_xsplitter vout display error: Failed to open Xsplitter:mmal_vout module
> [0000007f700013e0] mmal_xsplitter vout display: Try drm
> [0000007f70001fc0] drm_vout generic: <<< OpenDrmVout: Fmt=I420
> [0000007f70001fc0] drm_vout generic error: Failed to get xlease
> 
> 
> ^C[mpeg4 @ 0x7f7cc28d70] get_buffer() failed
> [mpeg4 @ 0x7f7cc28d70] thread_get_buffer() failed
> [mpeg4 @ 0x7f7cc28d70] get_buffer() failed (-12 (nil))
> ```

Definitely sounds like some permission is missing then.

File: ansible/roles/system/tasks/main.yml [hunk 2]
```diff
@@ -52,12 +52,6 @@
     regexp: ^(dtoverlay=vc4-kms-v3d)$
     replace: '#\1'
 
-- name: Add FKMS config for Pi 4.
-  ansible.builtin.lineinfile:
-    path: /boot/config.txt
-    insertafter: '^\[pi4\]$'
-    line: dtoverlay=vc4-fkms-v3d
```
Comment (nicomiguelino) [inline ansible/roles/system/tasks/main.yml:59]: Bing back these lines...

**Comment by nicomiguelino** [comment]
@vpetersson, I'd like to request for a review...

**Review by vpetersson — APPROVED** [review]

**Pull request merged**